torch/_dynamo/guards.py

```diff
@@ -342,11 +343,19 @@ def DICT_KEYS(self, guard):
         code.append(f"___check_type_id({ref}, {self.id_ref(t)})")
         param_key_ids = set(dict_param_key_ids(value))
         const_keys = set(dict_const_keys(value))
+        if any(isinstance(k, enum.Enum) for k in const_keys):
+            # Workaround repr(Enum) returns invalid global reference before python 3.11
+            # https://peps.python.org/pep-0663/
+            const_keys_str = f"{set([str(k) if isinstance(k, enum.Enum) else repr(k) for k in const_keys])}".replace(
```

Comment:
Could this break if someone has a string with the same name as the enum? (could be rare but thought it's worth pointing out).

To rectify this, you could introduce a `___dict_enum_keys` helper which filters out the enum keys into a set of strings and store a {enum_keys_str} in the guard code.

Reply:
Yes, it's rare but possible. I don't prefer to have a dedicated code to check the enum keys, we can filter out enum keys vs non-enum keys and generate two sets separately, and send the combined keys to codegen.